File: Merge Two 2D Arrays by Summing Values/Merge_Two_2D_Arrays_by_Summing_Values.py
class Solution:
    def mergeArrays(self, nums1: List[List[int]], nums2: List[List[int]]) -> List[List[int]]:
        ans = []
        i = j = 0
        n, m = len(nums1), len(nums2)
        
        while i < n and j < m:
            if nums1[i][0] < nums2[j][0]:
                current = nums1[i]
                i += 1
            else:
                current = nums2[j]
                j += 1
            if ans and ans[-1][0] == current[0]:
                ans[-1][1] += current[1]
            else:
                ans.append([current[0], current[1]])
        
        while i < n:
            current = nums1[i]
            if ans and ans[-1][0] == current[0]:
                ans[-1][1] += current[1]
            else:
                ans.append([current[0], current[1]])
            i += 1
        
        while j < m:
            current = nums2[j]
            if ans and ans[-1][0] == current[0]:
                ans[-1][1] += current[1]
            else:
                ans.append([current[0], current[1]])
            j += 1
        
        return ans
        

File: Merge Two 2D Arrays by Summing Values/Merge_Two_2D_Arrays_by_Summing_Values.java
class Solution {
    public int[][] mergeArrays(int[][] nums1, int[][] nums2) {
        List<int[]> ans = new ArrayList<>();
        int i = 0, j = 0;
        int n = nums1.length, m = nums2.length;
        while (i < n && j < m) {
            if (nums1[i][0] < nums2[j][0]) {
                if (ans.isEmpty() || ans.get(ans.size() - 1)[0] < nums1[i][0]) {
                    ans.add(new int[]{nums1[i][0], nums1[i][1]});
                } else {
                    ans.get(ans.size() - 1)[1] += nums1[i][1];
                }
                i++;
            } else {
                if (ans.isEmpty() || ans.get(ans.size() - 1)[0] < nums2[j][0]) {
                    ans.add(new int[]{nums2[j][0], nums2[j][1]});
                } else {
                    ans.get(ans.size() - 1)[1] += nums2[j][1];
                }
                j++;
            }
        }
        while (i < n) {
            if (ans.isEmpty() || ans.get(ans.size() - 1)[0] < nums1[i][0]) {
                ans.add(new int[]{nums1[i][0], nums1[i][1]});
            } else {
                ans.get(ans.size() - 1)[1] += nums1[i][1];
            }
            i++;
        }
        while (j < m) {
            if (ans.isEmpty() || ans.get(ans.size() - 1)[0] < nums2[j][0]) {
                ans.add(new int[]{nums2[j][0], nums2[j][1]});
            } else {
                ans.get(ans.size() - 1)[1] += nums2[j][1];
            }
            j++;
        }
        return ans.toArray(new int[ans.size()][]);
    }
}